server/matchmaker/search.py

```diff
@@ -234,7 +247,20 @@ def _players_repr(self) -> str:
 
     def __repr__(self) -> str:
         """For debugging"""
-        return f"Search({[p.login for p in self.players]})"
+        string = f"Search({[p.login for p in self.players]}, {self.average_rating}"
+        if self.failed_matching_attempts > 0:
+            string += f", FMA: {self.failed_matching_attempts}"
+        if self.has_newbie():
+            string += ", has_newbie"
+        string += ")"
```

Comment:
Pretty low hanging fruit but rewriting this so there's no `+=` on strings might make it better